newIDE/app/src/Profile/ContributionsDetails.js

```diff
@@ -0,0 +1,210 @@
+// @flow
+import { Trans } from '@lingui/macro';
+
+import * as React from 'react';
+import { Accordion } from '../UI/Accordion';
+import { AccordionHeader } from '../UI/Accordion';
+import { AccordionBody } from '../UI/Accordion';
+import { IconContainer } from '../UI/IconContainer';
+import { Column, Line } from '../UI/Grid';
+import {
+  type ExtensionShortHeader,
+  getUserExtensionShortHeaders,
+} from '../Utils/GDevelopServices/Extension';
+import {
+  type ExampleShortHeader,
+  getUserExampleShortHeaders,
+} from '../Utils/GDevelopServices/Example';
+import PlaceholderLoader from '../UI/PlaceholderLoader';
+import Text from '../UI/Text';
+
+type LineProps = {|
+  fullName: string,
+  previewIconUrl?: string,
+  shortDescription?: string,
+|};
+
+export const ContributionLine = ({
+  fullName,
+  previewIconUrl,
+  shortDescription,
+}: LineProps) => (
+  <Line>
+    {previewIconUrl && (
+      <IconContainer alt={fullName} src={previewIconUrl} size={64} />
+    )}
+    <Column expand>
+      <Text noMargin>{fullName}</Text>
+      <Text noMargin size="body2">
+        {shortDescription}
+      </Text>
+    </Column>
+  </Line>
+);
+
+type Props = {
+  userId: string,
+};
+
+export default ({ userId }: Props) => {
+  const [
+    extensions,
+    setExtensions,
+  ] = React.useState<?Array<ExtensionShortHeader>>(null);
+  const [examples, setExamples] = React.useState<?Array<ExampleShortHeader>>(
+    null
+  );
+  const [extensionError, setExtensionError] = React.useState<?Error>(null);
+  const [exampleError, setExampleError] = React.useState<?Error>(null);
+
+  const fetchUserExtensions = React.useCallback(
+    () => {
+      (async () => {
+        if (!userId) return;
+        try {
+          const extensions = await getUserExtensionShortHeaders(userId);
+          setExtensions(extensions);
+        } catch (error) {
+          console.error('Error while loading extensions:', error);
+          setExtensionError(error);
+        }
+      })();
+    },
+    [userId]
+  );
+
+  React.useEffect(
+    () => {
+      fetchUserExtensions();
+    },
+    [fetchUserExtensions]
+  );
+
+  const fetchUserExamples = React.useCallback(
+    () => {
+      (async () => {
+        if (!userId) return;
+        try {
+          const examples = await getUserExampleShortHeaders(userId);
+          setExamples(examples);
+        } catch (error) {
+          console.error('Error while loading examples:', error);
+          setExampleError(error);
+        }
+      })();
+    },
+    [userId]
+  );
+
+  React.useEffect(
+    () => {
+      fetchUserExamples();
+    },
+    [fetchUserExamples]
+  );
+
+  const ExtensionsAccordion = () => {
```

Comment:
Could you either move this to a different component outside this one (to avoid redeclaring new components at each render, which will make React diffing useless) or rename these to renderXxx with a React.useCallback?